### PR TITLE
Introduce addArguments and addJvmArguments methods on LongRunningOperation

### PR DESCRIPTION
**Pull request by donat**

### CODE REVIEW

File: subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java [hunk 1]
```diff
@@ -111,6 +111,16 @@
...
 
+    /**
+     * Appends one or more Java VM arguments to the existing list.
+     *
+     * @param jvmArgument the argument to use for the Gradle process
+     * @param moreJvmArguments more arguments to to use for the Gradle process
+     * @return this
+     * @since 5.0
+     */
+    LongRunningOperation addJvmArguments(@Nullable String jvmArgument, @Nullable String... moreJvmArguments);
```
Comment (oehme) [inline subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java]: ❌ Why the single argument + varargs?

File: subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java [hunk 2]
```diff
@@ -111,6 +111,16 @@
...
+
     /**
```
Comment (oehme) [inline subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java]: ❌ Missing overload taking `Iterable`

File: subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java (hunk 1 above)
Comment (oehme) [reply subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java]: ❌ Should not be nullable.

Comment (donat) [reply subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java]: I'll remove the varargs.

Comment (donat) [reply subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java]: It's just easier to specify multiple arguments at once, i.e. `operation.addArguments('--init-script', '/path-to-script')`.

Comment (donat) [reply subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java]: Re: nullable: I've only followed the pattern on the existing methods.

Comment (oehme) [reply subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java]: I don't understand what you mean. `operation.addArguments('--init-script', '/path-to-script') ` doesn't require this two-argument pattern. It should just take varargs. The first parameter doesn't add anything to this method.

The existing methods are setters, which allow you to say "I want you to use the default arguments" by passing `null`. That doesn't make sense for an `add` method.

Comment (donat) [reply subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java]: Ok, but if I remove the first parameter clients can call `addArguments()` without arguments which is a no-op.  I just wanted to remove that possibility.

Comment (donat) [reply subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java]: Ok, that makes sense. I didn't think about the use case when the client wants to pass through a collection.

File: subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java (hunk 2 above)
Comment (donat) [reply subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java]: Done

File: subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java (hunk 1 above)
Comment (donat) [reply subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java]: Done

**Pull request merged**